31_HLSLPathTracer/app_resources/hlsl/ray_gen.hlsl

```diff
@@ -0,0 +1,75 @@
+#ifndef _NBL_HLSL_EXT_RAYGEN_INCLUDED_
+#define _NBL_HLSL_EXT_RAYGEN_INCLUDED_
+
+#include <nbl/builtin/hlsl/sampling/box_muller_transform.hlsl>
+
+#include "common.hlsl"
+
+namespace RayGen
+{
+
+template<class Ray>
+struct Basic
+{
+    using this_t = Basic<Ray>;
+    using ray_type = Ray;
+    using scalar_type = typename Ray::scalar_type;
+    using vector3_type = typename Ray::vector3_type;
+    
```

Comment:
this looks useful put it in the `path_tracing` namespace and builtin directory as `DefaultRayGen` or `BasicRayGen`